### PR TITLE
PLFM-1409: Fix getAllVersions service

### PR DESCRIPTION
**Pull request by RadicalZephyr**

### CODE REVIEW

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/EntityManagerImpl.java [hunk 1]
```diff
@@ -133,6 +134,26 @@ public EntityManagerImpl(NodeManager nodeManager,
...
 	}
 
+	@Override
+	public VersionInfo getEntityVersionInfo(UserInfo userInfo, String entityId,
+			long versionNumber) throws DatastoreException, UnauthorizedException, NotFoundException {
+		Node node = nodeManager.getNodeForVersionNumber(userInfo, entityId, versionNumber);
+		VersionInfo info = new VersionInfo();
+		info.setId(node.getId());
+		info.setType(node.getNodeType());
+		info.setModifiedBy(userManager.getDisplayName(node.getCreatedByPrincipalId()));
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/EntityManagerImpl.java]: Needs to be the principal id.

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/EntityManagerImpl.java [hunk 2]
```diff
@@ -133,6 +134,26 @@ public EntityManagerImpl(NodeManager nodeManager,
 		return populateEntityWithNodeAndAnnotations(entityClass, annos, node);
 	}
 
+	@Override
+	public VersionInfo getEntityVersionInfo(UserInfo userInfo, String entityId,
+			long versionNumber) throws DatastoreException, UnauthorizedException, NotFoundException {
+		Node node = nodeManager.getNodeForVersionNumber(userInfo, entityId, versionNumber);
+		VersionInfo info = new VersionInfo();
+		info.setId(node.getId());
+		info.setType(node.getNodeType());
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/EntityManagerImpl.java]: Type is not part of the version info.

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/NodeDAOImpl.java [hunk 3]
```diff
@@ -147,12 +159,15 @@
...
 	private static String SQL_GET_ALL_VERSION_NUMBERS = "SELECT "+COL_REVISION_NUMBER+" FROM "+TABLE_REVISION+" WHERE "+COL_REVISION_OWNER_NODE +" = ? ORDER BY "+COL_REVISION_NUMBER+" DESC";
+
 	
 	private static String SQL_COUNT_ALL = "SELECT COUNT("+COL_NODE_ID+") FROM "+TABLE_NODE;
 	// Used to determine if a node id already exists
 	private static String SQL_COUNT_NODE_ID = "SELECT COUNT("+COL_NODE_ID+") FROM "+TABLE_NODE+" WHERE "+COL_NODE_ID +" = :"+BIND_ID_KEY;
 	private static String SQL_COUNT_REVISON_ID = "SELECT COUNT("+COL_REVISION_OWNER_NODE+") FROM "+TABLE_REVISION+" WHERE "+COL_REVISION_OWNER_NODE +" = ? AND "+COL_REVISION_NUMBER+" = ?";
-
+	private static String SQL_COUNT_REVISONS = "SELECT COUNT("
+			+ COL_REVISION_OWNER_NODE + ") FROM " + TABLE_REVISION + " WHERE "
```
Comment (RadicalZephyr) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/NodeDAOImpl.java]: Should be counting Number column

File: lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/dao/NodeDAOImplTest.java [hunk 4]
```diff
@@ -879,6 +880,19 @@ public void testGetVersionNumbers() throws Exception {
...
+		String id = createNodeWithMultipleVersions(numberVersions);
+		// Now list the versions
+		List<VersionInfo> versionsOfEntity = nodeDao.getVersionsOfEntity(id, 0, 10);
+		assertNotNull(versionsOfEntity);
+		assertEquals(numberVersions,versionsOfEntity.size());
+		assertEquals(new Long(numberVersions), versionsOfEntity.get(0).getVersionNumber());
+		assertEquals(new Long(1), versionsOfEntity.get(versionsOfEntity.size()-1).getVersionNumber());
+	}
+
+	@Test
```
Comment (RadicalZephyr) [inline lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/dao/NodeDAOImplTest.java]: Test the count sql too

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/EntityManagerImpl.java [hunk 5]
```diff
@@ -133,6 +134,20 @@ public EntityManagerImpl(NodeManager nodeManager,
 		return populateEntityWithNodeAndAnnotations(entityClass, annos, node);
 	}
 
+	@Override
+	public VersionInfo getEntityVersionInfo(UserInfo userInfo, String entityId,
+			long versionNumber) throws DatastoreException, UnauthorizedException, NotFoundException {
+		Node node = nodeManager.getNodeForVersionNumber(userInfo, entityId, versionNumber);
```
Comment (RadicalZephyr) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/EntityManagerImpl.java]: Remove this

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/EntityManagerImpl.java [hunk 6]
```diff
@@ -418,6 +433,21 @@ public EntityHeader getEntityHeader(UserInfo userInfo, String entityId)
...
 	@Override
+	public List<VersionInfo> getVersionsOfEntity(UserInfo userInfo, String entityId,
+			long offset, long limit) throws DatastoreException,
+			UnauthorizedException, NotFoundException {
+		// pass through
+		return nodeManager.getVersionsOfEntity(userInfo, entityId, offset, limit);
+	}
+
+	@Override
+	public long getVersionCount(String entityId)
```
Comment (RadicalZephyr) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/EntityManagerImpl.java]: These two methods should be one.  They should return a generic object which is a QueryResult<T> where T = VersionInfo.  Pattern this object off of EntityHeaderQueryResult (and replace that object with QueryResult<EntityHeader>).  Then, have the manager do user access checking.

**Pull request merged**